File: mcp-spike/jqassistant-mapping/src/main/java/org/slizaa/jqassistant/hierarchicalgraph/JQAssistant_MappingProvider.java
package org.slizaa.jqassistant.hierarchicalgraph;

import org.slizaa.hierarchicalgraph.graphdb.mapping.spi.IMappingProvider.DefaultMappingProvider;
import org.slizaa.hierarchicalgraph.graphdb.mapping.spi.annotations.SlizaaMappingProvider;

@SlizaaMappingProvider
public class JQAssistant_MappingProvider extends DefaultMappingProvider {

	public JQAssistant_MappingProvider() {
		super(IMappingProviderMetadata.createMetadata("org.slizaa.jqassistant.hierarchicalgraph",
				"Slizaa jQAssistant (hierarchical packages)", null, null),
				new JQAssistant_HierarchyProvider(), new JQAssistant_DependencyProvider(),
				new JQAssistant_LabelProvider(), new JQAssistant_NodeComparator(),
				new JQAssistant_NodeMetadataProvider());
	}
}

File: hierograph-mcp/jqassistant-mapping/src/main/kotlin/io/hierograph/mcp/jqa/hierarchicalgraph/JQAssistantMappingProvider.kt
package io.hierograph.mcp.jqa.hierarchicalgraph

import io.hierograph.hierarchicalgraph.graphdb.mapping.spi.DefaultMappingProvider
import io.hierograph.hierarchicalgraph.graphdb.mapping.spi.DefaultMappingProviderMetadata

class JQAssistantMappingProvider : DefaultMappingProvider(
    DefaultMappingProviderMetadata(
        identifier = "io.hierograph.jqassistant.hierarchicalgraph",
        name = "Hierograph jQAssistant (hierarchical packages)"
    ),
    JQAssistantHierarchyProvider(),
    JQAssistantDependencyProvider(),
    JQAssistantNodeMetadataProvider()
)
